client/src/layouts/header/searchForm/SearchForm.jsx

```diff
@@ -1,7 +1,36 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import SearchBar from './searchBar/SearchBar';
+import Cities from './cities/Cities';
+import classes from './searchForm.module.css';
+import { cities } from '../../../../arrayOfCities';
 
 function SearchForm() {
-  return <div></div>;
+  const [searchText, setSearchText] = useState('');
+  const [matchingCities, setMatchingCities] = useState([]);
+
+  const handleSearchText = e => {
+    setSearchText(e.target.value);
+  };
+
+  const findMatchingCities = () => {
+    const tempMatchingCities =
+      searchText !== ''
+        ? cities.filter(city => city.name.toLowerCase().startsWith(searchText.toLowerCase()))
+        : [];
+
```

Comment:
What do you filter here?

Reply:
when you start entering in searchBar this function will find matching cities

Reply:
Yeah.. I've understood. You are using mock data.